mkldnn.mk

```diff
@@ -0,0 +1,66 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+ifeq ($(USE_MKLDNN), 1)
+	MKLDNN_ROOTDIR = $(ROOTDIR)/3rdparty/mkldnn
```

Comment:
it would be nice if you could further eliminate some duplicate variables. also, why override the value here?

Reply:
Actually there is no duplicate variables here, MKLDNN_ROOTDIR is the mkldnn submodule root directory under 3rdparty/mkldnn, and other variables (like MKLDNN_BUILDDIR/MKLDNN_INSTALLDIR) are based on this variable. And MKLDNNROOT is the variable to point to the final generated library/header file root which will be used by mxnet directly. So no override here.

Reply:
Can the MKLDNNROOT be renamed to avoid confusion? The goal is that you don't have to explain this to the next 100 people who look at this file.